src/ng/directive/form.js

```diff
@@ -485,7 +485,7 @@ var formDirectiveFactory = function(isNgForm) {
             var controller = ctrls[0];
 
             // if `action` attr is not present on the form, prevent the default action (submission)
-            if (!('action' in attr)) {
+            if (!('action' in attr) && attr['method'] !== 'dialog') {
```

Comment:
What I meant is: if the action attribute is set and the method is set to dialog, your code change would not prevent the submit. Isn't `!('action' in attr) || attr['method'] !== 'dialog' what we want?